### PR TITLE
Call update-generated-protobuf from update-codegen

### PR DESCRIPTION
**Pull request by thockin**
Calling update-codegen.sh with no arguments runs all the functions in definition order.  Client-generation depends on protobuf, so protobuf codegen needs to be near the beginning.

Also add some debug output for protobuf generation.

Also hides some old, verbose debug output.

/kind cleanup
/sig architecture

```release-note
NONE
```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -83,6 +83,41 @@ fi
 # Code generation logic.
 #
 
+# protobuf generation
+#
+# Some of the later codegens depend on the results of this, so it needs to come
+# first in the case of regenerating everything.
+function codegen::protobuf() {
```
Comment (aojea) [inline hack/update-codegen.sh:90]: how these functions are executed? I can't see any caller

Comment (aojea) [reply hack/update-codegen.sh:90]: 😮‍💨 `list_codegens()`

File: hack/verify-generated-protobuf.sh [hunk 2]
```diff
@@ -1,65 +0,0 @@
...
-for APIROOT in ${APIROOTS}; do
-  TMP_APIROOT="${_tmp}/${APIROOT}"
-  echo "diffing ${APIROOT} against freshly generated protobuf"
-  ret=0
-  diff -Naupr -I 'Auto generated by' -x 'zz_generated.*' -x '.github' -x '.import-restrictions' "${KUBE_ROOT}/${APIROOT}" "${TMP_APIROOT}" || ret=$?
-  cp -a "${TMP_APIROOT}"/* "${KUBE_ROOT}/${APIROOT}/"
-  if [[ $ret -eq 0 ]]; then
-    echo "${APIROOT} up to date."
-  else
-    echo "${APIROOT} is out of date. Please run hack/update-generated-protobuf.sh"
```
Comment (aojea) [inline hack/verify-generated-protobuf.sh:62]: how is the protobug generation verified now? it seems it passes th flag `verify-only` but I can't see it used in the new function

Comment (thockin) [reply hack/verify-generated-protobuf.sh:62]: We make a worktree, remove all generated files, re-generate, and diff.  It's far cleaner and more consistent.  Many tools do not have a `--verify` flag.

Comment (thockin) [reply hack/verify-generated-protobuf.sh:62]:
```
$ ./hack/verify-codegen.sh protobuf
+++ [0122 09:24:54] Generating protobufs for 66 targets
+++ [0122 09:24:54] Verifying Prerequisites....
+++ [0122 09:24:55] Building Docker image kube-build:build-aeca35f1ba-5-v1.26.0-go1.19.5-bullseye.0
+++ [0122 09:24:57] Creating data container kube-build-data-aeca35f1ba-5-v1.26.0-go1.19.5-bullseye.0
+++ [0122 09:25:10] Syncing sources to container
+++ [0122 09:25:14] Output from this container will be rsynced out upon completion. Set KUBE_RUN_COPY_OUTPUT=n to disable.
+++ [0122 09:25:14] Running build command...
+++ [0122 09:25:41] Syncing out of container
Not currently on any branch.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   staging/src/k8s.io/api/networking/v1/generated.pb.go
	modified:   staging/src/k8s.io/api/networking/v1/generated.proto

no changes added to commit (use "git add" and/or "git commit -a")
diff --git a/staging/src/k8s.io/api/networking/v1/generated.pb.go b/staging/src/k8s.io/api/networking/v1/generated.pb.go
index e9566d57e27..9c86b6c966b 100644
--- a/staging/src/k8s.io/api/networking/v1/generated.pb.go
+++ b/staging/src/k8s.io/api/networking/v1/generated.pb.go
@@ -868,114 +868,115 @@ func init() {
 }
 
 var fileDescriptor_1c72867a70a7cc90 = []byte{
-       // 1715 bytes of a gzipped FileDescriptorProto

<elided for this comment>

Generated files need to be updated
Please run 'hack/update-codegen.sh'
```

Comment (aojea) [reply hack/verify-generated-protobuf.sh:62]: I'm trying to find where that magic happens 😄

Comment (thockin) [reply hack/verify-generated-protobuf.sh:62]: verify-codegen.sh

```
_tmpdir="$(kube::realpath "$(mktemp -d -t "$(basename "$0").XXXXXX")")"
git worktree add -f -q "${_tmpdir}" HEAD
kube::util::trap_add "git worktree remove -f ${_tmpdir}" EXIT
cd "${_tmpdir}"

# Update generated code
export UPDATE_API_KNOWN_VIOLATIONS=true
hack/update-codegen.sh "$@"

# Test for diffs
diffs=$(git status --porcelain | wc -l)
```

File: hack/update-codegen.sh (hunk 1 above)
Comment (thockin) [reply hack/update-codegen.sh:90]:
```
for g in "${codegens_to_run[@]}"; do
    "codegen::${g}" "${flags_to_pass[@]}"
done     
```

File: hack/verify-generated-protobuf.sh (hunk 2 above)
Comment (aojea) [reply hack/verify-generated-protobuf.sh:62]: duh , hahah, I had an old version checked out

**Pull request merged**